### DISCUSSION TITLE
Why the queries are not required?

### DISCUSSION THREAD

**Discussion by Developeranees** [ecyrbe/zodios#373] [Q&A] [answered]
<img width="909" alt="Screenshot 2023-03-14 at 6 21 13 PM" src="https://user-images.githubusercontent.com/43973272/225013887-f27edffc-fd00-4145-a864-aee313ad973d.png">
I created this api with params but when i call it is not showed required in ts.
<img width="1380" alt="Screenshot 2023-03-14 at 6 22 02 PM" src="https://user-images.githubusercontent.com/43973272/225014100-2f5a87a4-def4-419b-a966-3ec5171fcc25.png">

is there any reason ? it is also happening with headers as well like when type is header.

**Comment by ecyrbe** [comment]
can you share:
- your `package.json`
- your `tsconfig.json`

Or better provide a reproductible setup in stackblitz, codesandbox or simply a repository ?

here a working example in [tsplayground](https://www.typescriptlang.org/play?alwaysStrict=false#code/JYWwDg9gTgLgBAbxAQwNYFMCCZgBo4BaEAJsBAM4C+cAZlBCHAEQACAXiWeQPQDG06JgG4AsAChQkWIjZxqdBnADkHYktFjx-AHbl4yHHAC8cFBmzAAFAG0E4uKfQwAFiQBcygOZOlue3GQAG2Bkcg8lbxgAFRIIX38wZBdw7hhYnjdgNT8xB0SoZBAndCgw638HO1yHGu1C9HCAR3jqmrgYAE8wBuUARQBXEo6WtodyXmd0FA82ADo9KGBtT0sASgq5HLaq0bg6ovDE7xHRzu7wgaGTtvHJ6bg5haWV1dmIMBgyOsC1jcp-AC6WzgUHQ5Eguh6cwgACMAFboXgwSw7Gr9cglACSxBms20-RAMJKa2BDiyuPxhOJq1J7WAMECUPmMEWyxJG344EZMHQOIesxhEAgjOQ2nZ1Uo6zElABqw0WggungvGC6G08BM2nQAHdCJwKJYlM4YDAwGFuNw4eRFWBAsheOhXIFiCVZmdgPwXbNOb4Ajg5eItKr1bNIjFiBAUY1BoswR4EJRJRoLbtUw4AHoAfiAA)

**Comment by Developeranees** [comment, marked as answer]
Thanks for your reply 
Repository => https://github.com/Developeranees/zodios-issue.git

**Reply by ecyrbe** [reply]
your example is working fine on my side.
See it running on stackblitz :
https://stackblitz.com/github/Developeranees/zodios-issue?file=index.ts

So my guess is that your IDE could be using an incompatible typescript version ?
If so, you can tell your IDE to use the project TS version (example with vscode):
![image](https://user-images.githubusercontent.com/633115/225267202-a0f39421-e9f2-487a-ab9a-80f5736d8f5f.png)
![image](https://user-images.githubusercontent.com/633115/225267348-8fa98e5a-b46e-4fe0-836a-e564005b4d1a.png)

**Reply by Developeranees** [reply]
Oh Thank You Brother.
My Vs code is using nighty version => typescript 5.0...
thats causing the issue.
